### PR TITLE
Fix issues related to parallel build

### PR DESCRIPTION
**Pull request by orhan-mehmedov**
Add missing dependencies.
Add nproc parameter for make -j option.

Related to [APPLINK-25547](https://adc.luxoft.com/jira/browse/APPLINK-25547).

### CODE REVIEW

**Comment by orhan-mehmedov** [comment]
Please, review
@vlantonov, @AGaliuzov, @LevchenkoS, @anosach-luxoft, @dev-gh, @VVeremjova

**Comment by vlantonov** [comment]
Concerning the failure in CI build it is an issue I am going to fix with my next pull request. It is improper assignment of `const char*` variable.

File: .travis.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ before_install:
 script:
   - sudo ln -sf /usr/bin/gcov-4.9 /usr/bin/gcov
   - bash -e tools/infrastructure/check_style.sh
-  - mkdir build && cd build && cmake ../ -DBUILD_TESTS=ON -DENABLE_GCOV=ON && make install
+  - mkdir build && cd build && cmake ../ -DBUILD_TESTS=ON -DENABLE_GCOV=ON && echo nproc is $("nproc") && make install -j$("nproc")
```
Comment (LuxoftAKutsan) [inline .travis.yml]: Please extract  ```echo nproc is $("nproc")``` to separate line. And probably write it in more human readable. 
```
- ...
- echo " Number of processing units available :  $("nproc")"
- mkdir build && cd build  ...
```

File: src/components/formatters/CMakeLists.txt [hunk 2]
```diff
@@ -43,8 +43,15 @@ set(PATHS
   ${CMAKE_CURRENT_SOURCE_DIR}/src
 )
 
+set(LIBRARIES
+  protocol_handler
```
Comment (pestOO) [inline src/components/formatters/CMakeLists.txt]: Why formatter has a PH dependency?
Looks strange.

File: .travis.yml (hunk 1 above)
Comment (orhan-mehmedov) [inline .travis.yml]: @LuxoftAKutsan done [e5d6b2d](https://github.com/smartdevicelink/sdl_core/pull/856/commits/e5d6b2d6a5db9843dcd7dc5ef589afde3953cae2)

**Pull request merged**